packages/sdk/src/server/index.ts

```diff
@@ -7,37 +7,70 @@ import {
   ClientCredentials,
 } from "simple-oauth2";
 
+export type PipedreamOAuthClient = {
+  clientId: string;
+  clientSecret: string;
+};
+
 /**
  * Options for creating a server-side client.
  * This is used to configure the ServerClient instance.
  */
 export type CreateServerClientOpts = {
   /**
+   * @deprecated Use the `oauth` object instead.
    * The public API key for accessing the service.
    */
   publicKey?: string;
 
   /**
+   * @deprecated Use the `oauth` object instead.
    * The secret API key for accessing the service.
    */
   secretKey?: string;
 
   /**
+   * The environment in which the server client is running (e.g., "production", "development").
+   */
+  environment?: string;
+
+  /**
+   * @deprecated Use the `oauth` object instead.
    * The client ID of your workspace's OAuth application.
    */
   oauthClientId?: string;
 
   /**
+   * @deprecated Use the `oauth` object instead.
    * The client secret of your workspace's OAuth application.
    */
   oauthClientSecret?: string;
 
+  /**
+   * The OAuth object, containing client ID and client secret.
+   */
+  oauth?: PipedreamOAuthClient;
+
   /**
    * The API host URL. Used by Pipedream employees. Defaults to "api.pipedream.com" if not provided.
    */
   apiHost?: string;
+
+  /**
+   * Base domain for workflows. Used for custom domains: https://pipedream.com/docs/workflows/domains
+   */
+  baseWorkflowDomain?: string;
 };
 
+/**
+ * Different types of ways customers can authorize requests to HTTP endpoints
+ */
+export enum HTTPAuthType {
+  None = "none",
+  StaticBearer = "static-bearer",
```

Comment:
Should we keep these values consistent with the enums that we traffic[here](https://github.com/PipedreamHQ/pipedreamin/blob/master/api/app/models/http_interface.rb#L27)?

Reply:
yes good call, fixed